5 - RPG/Code/RPG.CharacterBuilder/CharacterAbilityClasses/Charisma.cs

```diff
@@ -0,0 +1,24 @@
+﻿namespace RPG.CharacterBuilder.CharacterAbilityClasses
+{
+    public class Charisma : CharacterPropertyBase<int>
+    {
+        private static int _defaultValue = 10;
+        public override int DefaultValue { get; internal set; } = _defaultValue;
+        public Charisma() : base(_defaultValue)
+        {
+
+        }
+        public Charisma(int value) : base(value)
+        {
+
+        }
+
+        public override bool IsValid(int value)
+        {
+            if (value >= 0 && value <= 20)
+                return true;
+            else
+                return false;
+        }
```

Comment:
If you look at all the abilities, they are all copy and paste of each other except for 2 things; Minimum supported value and Maximum supported value. Overall you have some inheritance and your design is clean and easy to read but I think given your current design you could push *more* into your base class so you have an ability specific base class. Then each class would simply need to specify a Min, Max, Default value. Would really boil down the code to what is unique and then the Validate method could exist in a baseclass, only ever needed once.